### DISCUSSION TITLE
zkSync Node Beta

### DISCUSSION THREAD

**Discussion by agolajko** [zkSync-Community-Hub/zksync-developers#526] [Announcements]
# Releasing the Beta version of our zkSync Node with a much simplified setup process!

Ahead of our decentralization we are taking the first step to provide access to our network by making it easy for anyone to monitor network activity via our Node.

The zkSync Node is a read-replica of the Main node that external parties can run. It functions by fetching data from the zkSync API and re-applying transactions locally, starting from the last block serving as a snapshot (complete network view of the main node at a given block for the purpose of syncing)— the zkSync Node shares most of its codebase with the Main node. Consequently, when it re-applies transactions, it does so exactly as the Main node did in the past. With this update developers will have low latency access to the zkSync network’s data for testing purposes as well as building third-party apps.

- This update moves the status of the zkSync Node from Alpha to Beta with features massively improving developer experience.
    - One line Docker setup versus a long installation process with the previous versions
    - Allows syncing from snapshots (as defined above) and staying up to date via our Main node
    - Running our zkSync Node via snapshots (as defined above) is now possible on consumer hardware

With the above update, the CPU and RAM requirements of the zkSync Node have been drastically decreased.

Those are requirements for a freshly started node and the the state grows about 1TB per month for mainnet

- 32 GB of RAM and a relatively modern CPU
- 30 GB of storage for testnet nodes
- 300 GB of storage for mainnet nodes
- 100 Mbps connection (1 Gbps+ recommended)

We will follow up with future releases that reduce storage space requirements by introducing Pruning (the process of erasing older data to save disk space). 

**Follow our guide for the setup in our [Docs](https://docs.zksync.io/infra/quickstart.html)**

**Comment by Xcertik-Realist** [comment]
add a quick start guide also maybe for less tech minded users ??

**Reply by Trional0** [reply]
Node running and operators shouldn't be non tech minded users and they have to know what they do.

**Reply by ajkol** [reply]
Nice inclusivity, good look for decentralization proponent

**Reply by 0xsisyfos** [reply]
Hey @Xcertik-Realist 
You can find the quickstart here: https://docs.zksync.io/infra/quickstart.html

**Comment by 8gh8** [comment]
good news)

**Comment by kingdjali** [comment]
Make a post for those of us with less tech experience

**Comment by Va1Ke** [comment]
FATAL:  the database system is starting up
dependency failed to start: container docker-compose-examples-postgres-1 is unhealthy

**Reply by tomg10** [reply]
Hi, what version of docker compose are you using?

**Reply by hatemosphere** [reply]
@Va1Ke could you please provide further debug information via:

```
docker ps --filter "name=docker-compose-examples-postgres-1"
docker logs docker-compose-examples-postgres-1
```

**Comment by skyc1e** [comment]
what's the cpu and gpu minimum requirements?

**Reply by rocketport** [reply]
https://docs.zksync.io/infra/quickstart.html#important-notes

**Comment by rocketport** [comment]
i see graphana in docker image - how to use it? 3000 port is closed. node work ok.

**Reply by tomg10** [reply]
Hi, it should be accessible under port 3000.
Please check the what happened to that container using `docker container inspect docker-compose-examples-grafana-1` or `docker container logs docker-compose-examples-grafana-1`

**Comment by Betamavro** [comment]
When can I register & join as Zksync Node Operator ? Thanks

**Reply by lemdev0** [reply]
hey, any info of a command line to get the keys?

**Comment by ozzyxbt** [comment]
> ERROR: Invalid interpolation format for "healthcheck" option in service "postgres": "psql -U postgres -c "select exists (select * from pg_stat_activity where datname = '{{ database_name }}' and application_name = 'pg_restore')" | grep -e ".f$""

Getting this error when I try to launch the node on my Ubuntu server.

**Reply by hatemosphere** [reply]
@OZNSN this should be fixed in https://github.com/matter-labs/zksync-era/pull/2148
please update to latest `main` branch and try again

**Comment by cyrptoburrys** [comment]
What i do after this? 

![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/136279935/12727069-89bd-4c86-a26f-611362374fbb)
